src/shared/varlink.c

```diff
@@ -1799,19 +1867,36 @@ void* varlink_get_userdata(Varlink *v) {
         return v->userdata;
 }
 
-static int varlink_acquire_ucred(Varlink *v) {
-        int r;
+void varlink_ucred_enable_impersonate(Varlink *v_send, Varlink *v_recv) {
```

Comment:
just call this varlink_impersonate(), that's expressive enough.

please add:

```c
assert_return(v_send, -EINVAL);
assert_return(v_recv, -EINVAL);
```

we generally do superficial validation of all function parameters.